driver.go

```diff
@@ -71,6 +72,9 @@ func (d MySQLDriver) Open(dsn string) (driver.Conn, error) {
 
 	// Enable TCP Keepalives on TCP connections
 	if tc, ok := mc.netConn.(*net.TCPConn); ok {
+		timeout := time.Now().Add(mc.cfg.timeout)
+		tc.SetReadDeadline(timeout)
+		tc.SetDeadline(timeout)
```

Comment:
It's total bad place to set deadline.
Additionaly, why do you set both of ReadDeadline and Deadline?